### PR TITLE
Added "Uninstall" Button in DT_Extensions Menu

### PR DESCRIPTION
**Pull request by prykon**
Now, if a plugin is installed but not active, the user can uninstall it from the DT Extensions menu.

uninstall button

This solves the following issue:
https://github.com/DiscipleTools/disciple-tools-theme/issues/1530

### CODE REVIEW

File: dt-core/admin/menu/tabs/tab-featured-extensions.php [hunk 1]
```diff
@@ -193,40 +216,47 @@ public function box_message( $tab ) {
...
                                 ?>
-                                <button class="button"
-                                        onclick="activate('<?php echo esc_html( $result_name ); ?>')"><?php echo esc_html__( 'Activate', 'disciple_tools' ) ?></button>
+                                    <li>
+                                        <button class="button" onclick="activate('<?php echo esc_html( $result_name ); ?>')"><?php echo esc_html__( 'Activate', 'disciple_tools' ) ?></button>
+                                    </li>
+                                    <li>
+                                        <button class="button" onclick="uninstall('<?php echo esc_html( $plugin->slug ); ?>/<?php echo esc_html( $plugin->slug );?>.php')"><?php echo esc_html__( 'Uninstall', 'disciple_tools' ) ?></button>
+                                        <!-- <a class="button" href="http://localhost:10033/wp-admin/plugins.php?action=delete-selected&checked%5B0%5D=disciple-tools-demo-content%2Fdisciple-tools-demo-content.php&plugin_status=all&paged=1&s&_wpnonce=0d0ef6e4f1"><?php echo esc_html__( 'Uninstall', 'disciple_tools' ) ?></a> -->
```
Comment (prykon) [inline dt-core/admin/menu/tabs/tab-featured-extensions.php:239]: Remove comment

File: dt-core/admin/menu/tabs/tab-featured-extensions.php [hunk 2]
```diff
@@ -115,18 +126,30 @@ public function box_message( $tab ) {
...
+            && check_ajax_referer( 'portal-nonce', sanitize_key( $_POST['_ajax_nonce'] ) )
+            && ( ( is_multisite() && is_super_admin() ) || ( ! is_multisite() && current_user_can( 'manage_dt' ) ) ) ) {
             //check for admin or multisite super admin
             //install plugin
-            $this->install_plugin( sanitize_text_field( wp_unslash( $_POST["install"] ) ) );
+            $this->install_plugin( sanitize_text_field( wp_unslash( $_POST['install'] ) ) );
+            exit;
+        }
+
+        elseif ( isset( $_POST ) && isset( $_POST['uninstall'] ) && is_admin() && isset( $_POST['_ajax_nonce'] )
```
Comment (prykon) [inline dt-core/admin/menu/tabs/tab-featured-extensions.php:138]: `isset( $_POST )` is redundant

File: dt-core/admin/menu/tabs/tab-featured-extensions.php [hunk 3]
```diff
@@ -115,18 +126,30 @@ public function box_message( $tab ) {
             //activate the plugin
             activate_plugin( sanitize_text_field( wp_unslash( $_POST["activate"] ) ) );
             exit;
-        } elseif ( isset( $_POST ) && isset( $_POST["install"] ) && is_admin() && isset( $_POST["_ajax_nonce"] )
-            && check_ajax_referer( 'portal-nonce', sanitize_key( $_POST["_ajax_nonce"] ) )
-            && ( ( is_multisite() && is_super_admin() ) || ( ! is_multisite() && current_user_can( "manage_dt" ) ) ) ) {
+        } elseif ( isset( $_POST ) && isset( $_POST['install'] ) && is_admin() && isset( $_POST['_ajax_nonce'] )
```
Comment (prykon) [inline dt-core/admin/menu/tabs/tab-featured-extensions.php:129]: `isset( $_POST )` is redundant

**Comment by prykon** [comment]
Updated logic...

**On regular sites:**
- Everything stays the same.

**On multisites:**
- If not installed, 'network install' button shows
- If installed, 'network uninstall' button shows
- If installed and deactivated, 'network activate' and 'single activate' buttons show
- If installed and activated, 'network deactivate' and 'single deactivate' buttons show
- 'Network activate' button takes user to network plugins admin page
- 'Network deactivate' button takes user to network plugins admin page

See screenshot for reference.
Screen Shot 2022-03-02 at 16 26 50

**Comment by corsacca** [comment]
@prykon 
This look busy.
image

On multisites, let's just keep it how it was: install/activate/deactivate

**Comment by prykon** [comment]
@corsacca This would mean DT extensions wouldn't be able to be turned on across the board for a multisite (at least from this page). Should I still roll it back?

**Comment by corsacca** [comment]
> DT extensions wouldn't be able to be turned on across the board for a multisite on a subsite.
Correct

**Comment by prykon** [comment]
Todo:
- git reset 5d161d027e71a9f37b3d4bca0d201c1759b3f57c
- git revert c5b1519fed104cbb77755191c9ce400d7fb6d368 # (PHPCS Passed commit)

**Comment by prykon** [comment]
Now normal admins can't install/uninstall plugins for their multisite subsite. They can, however, activate/deactivate them.

**Pull request merged**